.github/workflows/development.yml

```diff
@@ -0,0 +1,43 @@
+name: Development
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python:
+          - "3.12"
+          - "3.8"
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        run: pip install tox
+      - name: Run unit tests
+        run: tox -e test-unit -- -m "smoke or sanity"
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python:
+          - "3.12"
+          - "3.8"
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        run: pip install tox
+      - name: Run integration tests
+        run: tox -e test-integration -- -m smoke
```

Comment:
What do you think about leaving the `tox` only for local runnings and **_"python versions matrix"_** for GitHub Actions?

Reply:
I'm not sure I'm following. tox is there for local runs and GitHub actions to solidify all pieces behind the same tool rather than having separations for Makefile and tox. The tox config itself has the python environments still within it, GitHub actions requires the version matrix, though, to run on each python version and ensure the setup action installs everything correctly. The other benefit of this pathway that GitHub Actions recommends is that we have separate job reports coming in for each Python version that is being tested and, therefore, can quickly figure out what is broken and in what environment.

For more info, see GitHub's docs here: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#running-tests-with-tox

Reply:
Oh, thanks. Didn't know that Github works concurrently with tox. This is why I wonder :)